[DX-1353, TT-12172] Remove OIDC pages from docs

rewsmith opened the pull request:
### **User description**
[DX-1353](https://tyktech.atlassian.net/browse/DX-1353)
<br>

---

### Description
The OIDC auth mode very often leads to confusion with potential customers since it does not follow the standard.  We believe it would be better to simply use the JWT auth mode for OIDC as well.

There are 2 OIDC pages identified under this PR which HAVE NOT been removed.  Rather, they have been added to the deprecated pages in menu.yaml instead.  Those pages are:

[Worked Example - API with OpenIDC Using Auth0](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/advanced-configuration/integrate/api-auth-mode/oidc-auth0-example/) and [Integrate with OIDC](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/advanced-configuration/integrate/api-auth-mode/open-id-connect)

The following content files have been updated to remove links to the 2 OIDC pages that should be deprecated
- [External JWT integration](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/advanced-configuration/integrate/api-auth-mode/json-web-tokens)
- [SSO](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/advanced-configuration/integrate/sso#sso---the-generic-use-case#sso---the-generic-use-case)
- [Basic config OIDC](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/basic-config-and-security/security/authentication-authorization/openid-connect)
- [Dashboard release notes 2.9](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-dashboard/release-notes/archived-releases/version-2.9)
- [Gateway release notes 2.9](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/release-notes/archived-releases/version-2.9)


- [Integrate with OIDC page contains links to soft deprecated page for Auth0 example](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/advanced-configuration/integrate/api-auth-mode/open-id-connect)

[DX-1353]: https://tyktech.atlassian.net/browse/DX-1353?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ


___

### **PR Type**
documentation


___

### **Description**
- Updated the JWT documentation to include references and instructions for integrating with OIDC.
- Modified the menu configuration to reflect the deprecation of standalone OIDC pages and their integration under JWT.
- Ensured that the OIDC pages are still accessible under a deprecated section in the menu.


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>json-web-tokens.md</strong><dd><code>Update JWT documentation to include OIDC integration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/advanced-configuration/integrate/api-auth-mode/json-web-tokens.md
<li>Updated title to include OIDC.<br> <li> Added references to OIDC in the context of JWT integration.<br> <li> Clarified the use of JWTs for OIDC implementation.<br>


</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4656/files#diff-380f0883743fe7d484f12adef0ed46ab7474e0a862486bd0797b907177b5e6c5">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Update menu to reflect OIDC deprecation and JWT integration</code></dd></summary>
<hr>

tyk-docs/data/menu.yaml
<li>Updated menu title to include OIDC with JWT.<br> <li> Moved OIDC-related pages to deprecated section.<br>


</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4656/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+9/-9</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

dcs3spp reviewed (changes requested):
Thanks @rewsmith. I have added preview links for the deprecated pages.  Maybe remove the content / link to the examples from the 2 files to prevent users from accessing directly from the website:

**oidc-auth0-example**
- tyk-docs/content/advanced-configuration/integrate/api-auth-mode/open-id-connect.md
- tyk-docs/content/basic-config-and-security/security/authentication-authorization/openid-connect.md

**open-id-connect**
- tyk-docs/content/advanced-configuration/integrate/sso.md
- tyk-docs/content/basic-config-and-security/security/authentication-authorization/json-web-tokens.md
- tyk-docs/tyk-docs/content/basic-config-and-security/security/authentication-authorization/openid-connect.md
- tyk-docs/tyk-docs/content/product-stack/tyk-dashboard/release-notes/archived-releases/version-2.9.md
- tyk-docs/content/product-stack/tyk-gateway/release-notes/archived-releases/version-2.9.md

Is the intent to undo the deprecation of the pages later in the future?  If there is no intent in future to undo the deprecation then we could remove the content file(s) add an alias to redirect to an existing URL to avoid 404 errors.

rewsmith commented:
> Thanks @rewsmith. I have added preview links for the deprecated pages.  Maybe remove the content / link to the examples from the 2 files to prevent users from accessing directly from the website:
> 
> **oidc-auth0-example**
> 
> * tyk-docs/content/advanced-configuration/integrate/api-auth-mode/open-id-connect.md
> * tyk-docs/content/basic-config-and-security/security/authentication-authorization/openid-connect.md
> 
> **open-id-connect**
> 
> * tyk-docs/content/advanced-configuration/integrate/sso.md
> * tyk-docs/content/basic-config-and-security/security/authentication-authorization/json-web-tokens.md
> * tyk-docs/tyk-docs/content/basic-config-and-security/security/authentication-authorization/openid-connect.md
> * tyk-docs/tyk-docs/content/product-stack/tyk-dashboard/release-notes/archived-releases/version-2.9.md
> * tyk-docs/content/product-stack/tyk-gateway/release-notes/archived-releases/version-2.9.md
> 
> Is the intent to undo the deprecation of the pages later in the future?  If there is no intent in future to undo the deprecation then we could remove the content file(s) add an alias to redirect to an existing URL to avoid 404 errors.

Thanks for the comments @dcs3spp!   I've removed or changed references in the files you mention.  Could you please double check that I've not missed anything!?!

I don't believe there is any intent to undo the deprecation of the pages in future.  I followed the [deprecation instructions in the docs bible](https://tyktech.atlassian.net/wiki/spaces/EN/pages/1888976900/The+Docs+Bible+-+Docs+Contributors+Guide#%5BhardBreak%5DDeprecated-pages).  Happy to take your lead on whether or not to remove content files and add aliases to redirect.  No idea how to add an alias to redirect - maybe you could take care of that part?  We could redirect to the JSON Web Tokens page: https://tyk.io/docs/basic-config-and-security/security/authentication-authorization/json-web-tokens/

dcs3spp commented:
> [Worked Example - API with OpenIDC Using Auth0](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/advanced-configuration/integrate/api-auth-mode/oidc-auth0-example/) and [Integrate with OIDC](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/advanced-configuration/integrate/api-auth-mode/open-id-connect)

Thanks @rewsmith for the suggestions and thanks for including the details of the pages that need to be deprecated.

I have pushed some changes to the PR branch that remove the pages and setup the redirect via aliases to [json-web-tokens](https://tyk.io/docs/basic-config-and-security/security/authentication-authorization/json-web-tokens/). To setup a [redirect](https://tyktech.atlassian.net/wiki/spaces/EN/pages/1888976900/The+Docs+Bible+-+Docs+Contributors+Guide#What-should-I-do-if-I-need-to-move-existing-content-in-the-file-structure%3F): in the front matter yaml header of the markdown file that represents the [destination or target of the redirect](https://raw.githubusercontent.com/TykTechnologies/tyk-docs/90078a6f4a721be45ceb5ae8836e64e7919e6842/tyk-docs/content/basic-config-and-security/security/authentication-authorization/json-web-tokens.md), we can add deleted content file paths to the aliases list.

When visiting the previews for [Worked Example - API with OpenIDC Using Auth0](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/advanced-configuration/integrate/api-auth-mode/oidc-auth0-example/) and [Integrate with OIDC](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/advanced-configuration/integrate/api-auth-mode/open-id-connect) a redirect should occur to [json-web-tokens](https://tyk.io/docs/basic-config-and-security/security/authentication-authorization/json-web-tokens/)



Once peer reviewed and peer approval given I will merge and release

andyo-tyk commented on tyk-docs/content/basic-config-and-security/security/authentication-authorization/openid-connect.md:
```diff
@@ -11,6 +11,6 @@ aliases:
 ---
 
-Tyk comes with support for OpenID Connect Identity Tokens provided by any standards compliant OIDC provider.
+Tyk comes with support for OpenID Connect provided by any standards compliant OIDC provider.
```
```suggestion
[OpenID Connect]({{< ref "https://openid.net/developers/how-connect-works" >}} (OIDC) builds on top of OAuth 2.0, adding authentication. You can secure your APIs on Tyk by integrating with any standards compliant OIDC provider using [JSON Web Tokens]({{< ref "basic-config-and-security/security/authentication-authorization/json-web-tokens" >}}) (JWTs).

JWTs offer a simple way to use the third-party Identity Provider (IdP) without needing any direct integration between the Tyk and 3rd-party systems.

To integrate a 3rd party OAuth2/OIDC IdP with Tyk, all you will need to do is ensure that your IdP can issue OAuth2 JWT access tokens as opposed to opaque tokens.

The client application authenticates with the IdP which then provides an access token that is accepted by Tyk. Tyk will take care of the rest, ensuring that the rate limits and quotas of the underlying identity of the bearer are maintained across JWT token re-issues, so long as the "sub" (or whichever identity claim you chose to use) is available and consistent throughout and the policy that underpins the security clearance of the token exists too.
```

andyo-tyk replied:
Add the following aliases to this page:
  - /advanced-configuration/integrate/api-auth-mode/oidc-auth0-example/
  - /advanced-configuration/integrate/api-auth-mode/open-id-connect
  - /advanced-configuration/integrate/api-auth-mode/json-web-tokens

dcs3spp replied:
updated

andyo-tyk commented on tyk-docs/content/advanced-configuration/integrate/api-auth-mode/json-web-tokens.md:
```diff
@@ -1,6 +1,6 @@
 ---
 date: 2017-03-24T16:40:31Z
-title: Integrate with JWT
+title: Integrate with JWT and OIDC
```
Delete this page

dcs3spp replied:
updated

andyo-tyk commented on tyk-docs/content/basic-config-and-security/security/authentication-authorization/json-web-tokens.md:
```diff
@@ -9,7 +9,9 @@ menu:
   - /security/your-apis/json-web-tokens/
-  - /tyk-apis/tyk-gateway-api/api-definition-objects/jwt/docs/basic-config-and-security/security/authentication-authorization/json-web-tokens/ 
+  - /tyk-apis/tyk-gateway-api/api-definition-objects/jwt/docs/basic-config-and-security/security/authentication-authorization/json-web-tokens/
+  - /advanced-configuration/integrate/api-auth-mode/oidc-auth0-example/
```
```suggestion
```

Move this alias to /basic-config-and-security/security/authentication-authorization/openid-connect.md

dcs3spp replied:
updated

andyo-tyk commented on tyk-docs/content/basic-config-and-security/security/authentication-authorization/json-web-tokens.md:
```diff
@@ -9,7 +9,9 @@ menu:
-  - /tyk-apis/tyk-gateway-api/api-definition-objects/jwt/docs/basic-config-and-security/security/authentication-authorization/json-web-tokens/ 
+  - /tyk-apis/tyk-gateway-api/api-definition-objects/jwt/docs/basic-config-and-security/security/authentication-authorization/json-web-tokens/
+  - /advanced-configuration/integrate/api-auth-mode/oidc-auth0-example/
+  - /advanced-configuration/integrate/api-auth-mode/open-id-connect
```
```suggestion
```
Move this alias to /basic-config-and-security/security/authentication-authorization/openid-connect.md

dcs3spp replied:
updated

andyo-tyk commented on tyk-docs/content/basic-config-and-security/security/authentication-authorization/openid-connect.md:
```diff
@@ -11,6 +11,6 @@ aliases:
+Tyk comes with support for OpenID Connect provided by any standards compliant OIDC provider.
 
-Getting OIDC support in the Dashboard requires only a few fields to be set up in the Core settings tab. for details see [Setting up OIDC]({{< ref "advanced-configuration/integrate/api-auth-mode/open-id-connect#setting-up-oidc" >}}) and a [worked example of using OpenID Connect with Auth0]({{< ref "advanced-configuration/integrate/api-auth-mode/oidc-auth0-example" >}}).
+Getting OIDC support in the Dashboard requires only JWT auth to be set up in the Core settings tab. For details see [JSON Web Tokens]({{< ref "basic-config-and-security/security/authentication-authorization/json-web-tokens" >}}).
```
```suggestion
```

dcs3spp replied:
updated

andyo-tyk commented on tyk-docs/data/menu.yaml:
```diff
@@ -1887,18 +1887,10 @@ menu:
               category: Page
               show: True
-            - title: "Integrate with JWT"
+            - title: "Integrate with JWT and OIDC"
```
Delete this page

dcs3spp replied:
updated

andyo-tyk reviewed (changes requested):
Proposed simplification - more work required on the menu structure, but that can follow in a separate PR.

dcs3spp commented:
I have updated so that the following pages have been deleted and now redirect to [OpenID Connect](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/basic-config-and-security/security/authentication-authorization/openid-connect) page:
1. [Auth0 Example](https://tyk.io/docs/advanced-configuration/integrate/api-auth-mode/oidc-auth0-example)
2. [OpenID Connect Example](https://tyk.io/docs/advanced-configuration/integrate/api-auth-mode/open-id-connect)
3. [3rd Party JWTs](https://tyk.io/docs/advanced-configuration/integrate/api-auth-mode/json-web-tokens)

Can test the redirects to [OpenID Connect](https://deploy-preview-4656--tyk-docs.netlify.app/docs/nightly/basic-config-and-security/security/authentication-authorization/openid-connect) using the links below:
1. [OIDC Auth0 Example](https://deploy-preview-4656--tyk-docs.netlify.app/docs/advanced-configuration/integrate/api-auth-mode/oidc-auth0-example/)
2. [OpenID Connect Example](https://deploy-preview-4656--tyk-docs.netlify.app/docs/integrate/api-auth-mode/open-id-connect)
3. [3rd Party JWTs](https://deploy-preview-4656--tyk-docs.netlify.app/docs/advanced-configuration/integrate/api-auth-mode/json-web-tokens)

I added the [suggested content](https://github.com/TykTechnologies/tyk-docs/pull/4656#discussion_r1611892944) to the [OpenID Connect](/docs/nightly/basic-config-and-security/security/authentication-authorization/openid-connect) page.

andyo-tyk reviewed (approved):
Thanks @dcs3spp

dcs3spp commented:
/release to release-5.3

dcs3spp commented:
/release to release-5.2

dcs3spp commented:
/release to release-5.1
